fili-sql/src/main/java/com/yahoo/bard/webservice/sql/DruidQueryToSqlConverter.java

```diff
@@ -929,6 +939,55 @@ private void sqlAggregationHelper(
         return allGroupBys;
     }
 
+    /**
+     * Find all word positions in string.
+     * @param textString a string to search into.
+     * @param word a target word.
+     * @return an array of indexes of all positions word was found in the textString.
+     */
+    private List<Integer> findWord(String textString, String word) {
+        List<Integer> indexes = new ArrayList<>();
+        String lowerCaseTextString = textString.toLowerCase(Locale.ENGLISH);
+        String lowerCaseWord = word.toLowerCase(Locale.ENGLISH);
+        int wordLength = 0;
+
+        int index = 0;
+        while (index != -1) {
+            index = lowerCaseTextString.indexOf(lowerCaseWord, index + wordLength);
+            if (index != -1) {
+                indexes.add(index);
+            }
+            wordLength = word.length();
+        }
+        return indexes;
+    }
+
+    /**
+     * Removes the quotes around the Sketch metric literal.
+     * @param sqlQuery the sql string built by the RelBuilder.
+     * @return the sql string without quote around the sketch metric expression.
+     */
+    private String formatSketchQuery(String sqlQuery) {
+        if (sqlQuery.contains(SKETCH_STRING)) {
+            StringBuilder stringBuilder = new StringBuilder();
+            int beginningPos = 0;
+            for (Integer pos : findWord(sqlQuery, SKETCH_STRING)) {
+                // Append string up to opening quote
+                stringBuilder.append(sqlQuery, beginningPos, pos);
+                beginningPos = pos + 1;
+                // Find closing quote location
+                int nextPos = beginningPos + sqlQuery.substring(beginningPos).indexOf("'");
+                // Append string up to closing quote
+                stringBuilder.append(sqlQuery, beginningPos, nextPos);
+                beginningPos = nextPos + 1;
+            }
+            stringBuilder.append(sqlQuery, beginningPos, sqlQuery.length());
+            return stringBuilder.toString();
+        } else {
```

Comment:
This whole method feels like it would have more sensibly be evaluated with regular expressions.